js/full-picture-render.js

```diff
@@ -14,41 +17,70 @@ const onDocumentKeydown = (evt) => {
-};
-
 const addBigPictureData = (photo) => {
   bigPicture.querySelector('.big-picture__img').querySelector('img').src = photo.url;
   bigPicture.querySelector('.likes-count').textContent = photo.likes;
   bigPicture.querySelector('.comments-count').textContent = photo.comments.length;
   bigPicture.querySelector('.social__caption').textContent = photo.description;
 };
 
-const addCommentData = (photo) => {
+const getCommentData = (photo) => {
   const commentsFragment = document.createDocumentFragment();
-  photo['comments'].forEach((comment) => {
-    const commentItem = commentsTemplate.cloneNode(true);
-    commentItem.querySelector('img').src = comment.avatar;
-    commentItem.querySelector('img').alt = comment.name;
-    commentItem.querySelector('p').textContent = comment.message;
-    commentsFragment.append(commentItem);
-  });
-  socialComments.innerHTML = '';
-  socialComments.append(commentsFragment);
+  let index = 0;
+  return (numberToShow) => {
+    for (let i = 0; i < numberToShow; i++) {
+      const comment = photo.comments[index];
+      const commentItem = commentsTemplate.cloneNode(true);
+      commentItem.querySelector('img').src = comment.avatar;
+      commentItem.querySelector('img').alt = comment.name;
+      commentItem.querySelector('p').textContent = comment.message;
+      commentsFragment.append(commentItem);
+      index++;
+    }
+    socialComments.append(commentsFragment);
+  };
 };
 
 const renderBigPicture = (evt, photo) => {
 
   evt.preventDefault();
   document.body.classList.add('modal-open');
 
-  hideElements();
+  socialComments.innerHTML = '';
+  const createComments = getCommentData(photo);
+  createComments(NUMBER_OF_VISIBLE_COMMENTS > photo.comments.length
+    ? photo.comments.length : NUMBER_OF_VISIBLE_COMMENTS);
 
-  addBigPictureData(photo);
+  const onCommentButtonClick = () => {
+    const socialCommentsCount = socialComments.children.length;
+    const numberToShow = photo.comments.length - socialCommentsCount < NUMBER_OF_VISIBLE_COMMENTS ?
+      photo.comments.length - socialCommentsCount : NUMBER_OF_VISIBLE_COMMENTS;
+    createComments(numberToShow);
+    bigPicture.querySelector('.social__comment-count').innerHTML =
+      `${socialComments.children.length} из <span class="comments-count">${photo.comments.length}</span> комментариев`;
+    if (socialComments.children.length === photo.comments.length) {
+      bigPicture.querySelector('.comments-loader').classList.add('hidden');
+      commentLoaderButton.removeEventListener('click', onCommentButtonClick);
+    }
+  };
+
+  const addCommentsCount = () => {
+    if (photo.comments.length <= NUMBER_OF_VISIBLE_COMMENTS) {
+      bigPicture.querySelector('.social__comment-count').innerHTML =
+        `${photo.comments.length} из <span class="comments-count">${photo.comments.length}</span> комментариев`;
+      bigPicture.querySelector('.comments-loader').classList.add('hidden');
```

Comment:
Давай вынесем в отдельные функции логику скрытия и показа кнопки

```suggestion
      const showCommentLoaderButton = (onCommentButtonClick) => {
          bigPicture.querySelector('.comments-loader').classList.remove('hidden');
          bigPicture.addEventListener('click', onCommentButtonClick);
      };
```